lib/xhr/XHRShim.js

```diff
@@ -0,0 +1,142 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, forin: true, maxerr: 50, regexp: true */
+/* global DOMParser */
+(function(global) {
+    "use strict";
+
+    // This script requires access to the transport to send
+    // file requests to the editor as commands
+    var transport = global._Brackets_LiveDev_Transport;
+
+    var XMLHttpRequest = global.XMLHttpRequest;
+    if(!XMLHttpRequest) {
+        return;
+    }
+
+    function sendMessage(msg) {
+        if(!transport || !transport.send) {
+            console.error("[Brackets LiveDev] No transport set");
+            return;
+        }
+
+        transport.send(JSON.stringify(msg));
+    }
+
+    function XMLHttpRequestLiveDev() {
+        var self = new XMLHttpRequest();
+        var requestUrl;
+        var abortCalled = false;
+
+        var open = self.open;
```

Comment:
One convention you might use here is to name your cached versions of methods/objects with a `$` suffix.  So you'd have `var $open = xhr.open`.  It helps to visually see that it's different.  Same below for other cases of this.